csharp/src/Ice/Reference.cs

```diff
@@ -1578,8 +1701,17 @@ internal void Write(OutputStream ostr)
             {
                 Debug.Assert(ostr.Encoding == Encoding.V20);
 
-                ostr.Write(Endpoints.Count > 0 ? ProxyKind.Direct : ProxyKind.Indirect);
-                ostr.WriteProxyData20(Identity, Protocol, Encoding, Location, InvocationMode, Facet);
+                ostr.Write(Endpoints.Count > 0 ? ProxyKind20.Direct :
+                    IsRelative ? ProxyKind20.IndirectRelative : ProxyKind20.Indirect);
+
+                IReadOnlyList<string> location = Location;
+                if (IsRelative && location.Count > 1)
+                {
+                    // Reduce location to its last segment
+                    location = ImmutableArray.Create(location[location.Count - 1]);
```

Comment:
Why only the last segment?

Reply:
See also the Read method. Currently it's usefull in this scenario:
- you are making a call on an indirect proxy, say `ice:icegrid/adapt1/A/B`
- the servant returns a proxy to an indirect-relative proxy in the same Locator domain, say `ice:ice:icegrid/adapt2/X/Y?relative=true`. This proxy could reference an Ice object in another server.
- during marshaling, `icegrid` gets chopped off as we never use it during unmarshaling
- during the unmarshaling of the response, the relative proxy is unmarshaled as: `ice:ice:icegrid/adapt2/X/Y`, i.e. we keep the domain (`location[0..^1]`) from the source proxy but replace its last segment using the relative proxy's location.

And later, the client's Locator resolves domain/adapt2, just like it resolved before domain/adapt1.

Note that currently locations are always 1 segment or empty, so it's of very limited use.